doc/2/guides/main-concepts/3-querying/index.md

```diff
@@ -8,3 +8,604 @@ order: 300
+This method efficiently mitigates the costs of scroll searches, but returns less consistent results: it's a middle ground, **ideal for real-time search requests**.  
+Also it's not possible to retrieve more than 10000 documents with this method.
+:::
+
+### Paginate with Scroll Cursor
+
+The `scroll` parameter can be specified in the search query to allows the usage of the [document:scroll](/core/2/api/controllers/document/scroll) action to **use a cursors to move through paginated results**.
+
+The **results from a scroll request are frozen**, and reflect the state of the collection at the time the initial search request.  
+For that reason, this action is **guaranteed to return consistent results**, even if documents are updated or deleted in the database between two pages retrieval.
+
+This is the **most consistent way to paginate results**, however, this comes at a **higher computing cost** for the server.
+
+To use this pagination method, you need to pass a `scroll` parameter with a duration. This duration corresponds to the **time during which Elasticsearch will keep your results frozen**. This duration will be refreshed at each call of the [document:scroll](/core/2/api/controllers/document/scroll) action.
+
+::: info
+The value of the `scroll` option should be the time needed to process one page of results.  
+This value has a maximum value which can be modified under the `services.storage.maxScrollDuration` [configuration](/core/2/guides/advanced/advanced/8-configuration) key.
+:::
+
+The search action will return a `scrollId` that you have to use with the [document:scroll](/core/2/api/controllers/document/scroll) to get the next page of results.
+
+**Example:** _Paginate search with scroll_
+```bash
+kourou sdk:query document:search \
+  -i ktm-open-data -c thamel-taxi -a scroll=30s -a size=2
+
+# {
+#   "hits": [
+#     {
+#       "_id": "aschen",
+#         # ...
+#       }
+#     },
+#     {
+#       "_id": "jenow",
+#         # ...
+#       }
+#     }
+#   ],
+#   "remaining": 2,
+#   "scrollId": "DXF1ZXJ5QW5kRmV0Y2gBAAAAAAAAACUWblZVaDV4UnNTck9mLU9wczZUVlBkUQ==",
+#   "total": 4
+# }
+```
+
+Then we have to pass the `scrollId` parameter to the [document:scroll](/core/2/api/controllers/document/scroll) action:
+```bash
+kourou sdk:query document:scroll -a scrollId=<scroll-id>
+
+# {
+#   "hits": [
+#     {
+#       "_id": "liia",
+#         # ...
+#       }
+#     },
+#     {
+#       "_id": "domisol",
+#         # ...
+#       }
+#     }
+#   ],
+#   "remaining": 0,
+#   "scrollId": "DXF1ZXJ5QW5kRmV0Y2gBAAAAAAAAACUWblZVaDV4UnNTck9mLU9wczZUVlBkUQ==",
+#   "total": 4
+# }
+```
+
+::: warning
+When using a cursor with the `scroll` option, Elasticsearch has to duplicate the transaction log to keep the same result during the entire scroll session.
+It **can lead to memory leaks** if a scroll duration too great is provided, or if too many scroll sessions are open simultaneously.
```

Comment:
```suggestion
It **can lead to memory leaks** if a scroll duration too great is provided, or if too many scroll sessions are open simultaneously.  

By default, Kuzzle sets a maximum scroll duration of 1 minute. This can be changed in the kuzzlerc configuration file.
```